airflow/models.py

```diff
@@ -1288,19 +1288,19 @@ def run(
             verbose=True)
 
         if not runnable and not mark_success:
-            if self.state != State.QUEUED:
-                # If a task's dependencies are met but it can't be run yet then queue it
-                # instead
-                self.state = State.QUEUED
-                msg = "Queuing attempt {attempt} of {total}".format(
-                    attempt=self.try_number % (task.retries + 1) + 1,
-                    total=task.retries + 1)
-                logging.info(hr + msg + hr)
-
-                self.queued_dttm = datetime.now()
-                msg = "Queuing into pool {}".format(self.pool)
-                logging.info(msg)
-                session.merge(self)
+            # we might have hit concurrency limits, which means we probably
+            # have been running prematurely.
+            self.state = State.UP_FOR_RETRY
```

Comment:
Yes, there are more issues around the concurrency of tasks. The pooling was seemingly fixed in 1.8, but concurrency follows a alternative path.

Ok, what I will do is for 1.8.0 set the state to None. I'm not happy with that, but I am comfortable enough with it that it won't affect the scheduler's behaviour.

I'll mark it TODO for 1.8.1 to fix it in a better way and will also adjust the logging message so that we are reminded to do so ;-). Agreed?